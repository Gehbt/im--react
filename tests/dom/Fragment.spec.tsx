// @vitest-environment jsdom

import IMReactDOM from "~core/im-react-dom";

suite("Fragment", () => {
  beforeAll(() => {
    vi.useFakeTimers({ toFake: ["requestIdleCallback"] });
    window.requestIdleCallback = vi.fn((cb: IdleRequestCallback) => {
      cb({
        timeRemaining: () => 12,
        didTimeout: false,
      });
      return 12;
    });
  });
  let root: HTMLElement;
  beforeEach(() => {
    vi.runAllTimers();
    root = (function () {
      const _div = document.createElement("div");
      _div.id = "app";
      return _div;
    })();
  });

  test("single Fragment", () => {
    IMReactDOM.createRoot(root).render(
      <>
        <p>Fragment</p>
        <p>Fragment2</p>
      </>,
    );
    expect(root).toMatchSnapshot();
  });

  test("single Fragment nested", () => {
    IMReactDOM.createRoot(root).render(
      <>
        <p>Fragment</p>
        <p>Fragment2</p>
        <>
          <p>FragmentL2</p>
          <p>Fragment2L2</p>
          <>
            <p>FragmentL3</p>
            <p>Fragment2L3</p>
          </>
        </>
      </>,
    );
    expect(root).toMatchSnapshot();
  });
});
