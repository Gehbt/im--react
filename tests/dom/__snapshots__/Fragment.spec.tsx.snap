// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fragment > single Fragment 1`] = `
<div
  id="app"
>
  <p>
    Fragment
  </p>
  <p>
    Fragment2
  </p>
</div>
`;

exports[`Fragment > single Fragment nested 1`] = `
<div
  id="app"
>
  <p>
    Fragment
  </p>
  <p>
    Fragment2
  </p>
  <p>
    FragmentL2
  </p>
  <p>
    Fragment2L2
  </p>
  <p>
    FragmentL3
  </p>
  <p>
    Fragment2L3
  </p>
</div>
`;
